adapt/engine.py

```diff
@@ -103,3 +105,279 @@ def register_intent_parser(self, intent_parser):
             self.intent_parsers.append(intent_parser)
         else:
             raise ValueError("%s is not an intent parser" % str(intent_parser))
+
+
+class DomainIntentDeterminationEngine(object):
+    """
+    DomainIntentDeterminationEngine.
+
+    The DomainIntentDeterminationEngine is a greedy and naive implementation of intent
+    determination. Given an utterance, it uses the Adapt parsing tools to come up with a
+    sorted collection of tagged parses. A valid parse result contains no overlapping
+    tagged entities in a single domain, and it's confidence is the sum of the tagged
+    entity confidences, which are weighted based on the percentage of the utterance
+    (per character) that the entity match represents.
+
+    This system makes heavy use of generators to enable greedy algorithms to short circuit
+    large portions of computation.
+    """
+
+    def __init__(self, tokenizer=None, trie=None, domain=0):
+        """
+        Initialize DomainIntentDeterminationEngine.
+
+        :param tokenizer: The tokenizer you wish to use.
+
+        :param trie: the Trie() you wish to use.
+
+        :param domain: a string representing the domain you wish to add
+        """
+        self.domains = {}
+        if tokenizer is not None or trie is not None:
+            self.register_domain(tokenizer=tokenizer, trie=trie, domain=domain)
+
+    @property
```

Comment:
Are there tests or samples that actually depend on this? It wasn't my intention to expose it, but apparently I was lazy and didn't prefix the property with __